library/cwm/src/lib/cwm/widget.rb

```diff
@@ -821,4 +821,72 @@ def replace_point
       ReplacePoint(Id(replace_point_id), VBox(VStretch(), HStretch()))
     end
   end
+
+  # Placeholder widget that is used to replace content on demand.
+  # The most important method is {replace} which allows switching content
+  class ReplacePoint < CustomWidget
+    # @param id [Object] id of widget. Needed to redefine only if more then one
+    # placeholder needed to be in dialog. Parameter type is limited by component
+    # system
+    # @param widget [CWM::AbstractWidget] initial widget in placeholder
+    def initialize(id: :_placeholder, widget: Empty.new(:_initial_placeholder))
+      self.handle_all_events = true
+      @id = id
+      @widget = widget
+    end
+
+    def contents
+      ReplacePoint(Id(@id), widget_content(@widget))
+    end
+
+    # Replaces content with different widget. All its events are properly
+    # handled.
+    # @param widget [CWM::AbstractWidget] widget to display and process events
+    def replace(widget)
+      log.info "replacing with new widget #{widget.inspect}"
+      Yast::UI.ReplaceWidget(@id, widget_content(widget))
+      @widget = widget
+      @widget.init if @widget.respond_to?(:init)
+    end
+
+    def help
+      @widget.respond_to?(:help) ? @widget.help : ""
```

Comment:
Likewise, `CWM::AbstractWidget` should rather always define a `help` returning `""`?